### ISSUE TITLE
GSoC 2016 Readme potential students

### ISSUE THREAD

**Issue by bphenriques** [systers/malaria-app-ios#32]
Hello there,

I contributed to PC malaria iOS mobile App last year mainly during GSoC 2015. I would like to leave some notes for future students who would like to join this project.

I posted some issues on this project's page to give a insight on what (in my opinion) can still be done in order to improve the application. However, the focus is to release the app and obtain feedback from the users therefore, at the moment, I think it is crucial to integrate with fabric.io to track crashes and have better insight on our users and fix the issue with infohub on iOS 9. Afterwards I suggest students to analyse the initial requirement documents at: https://github.com/systers/malaria-app-ios/blob/develop/InitialRequirements/RequirementsAugustApril2014.docx

As you can see, the most important requirements are:
- Gamification;
- Social Integration;
- Better integration with MACC (requirements are still up for debate since we depend on the server side)

Any other feature would be great but I suggest giving it a lower priority:
- Widgets
- Calendar integration

Non-critical features:
- Migrate to Realms would be cool but it is a user-centric requirement. I would look into it at the end of hte program.
- Refactor the storyboard and cartography before starting to add new features. Edit: the issue is postponed since it doesn't seem compatible with iOS 8.
  You can see the last year project presentation in the following link:
  https://docs.google.com/presentation/d/1yv4XuNNpTgDkyOjL9SSjGe7PSZuwpbah42mQCACxdXQ/edit?usp=sharing

In short... you can see that there is so much to do! But I suggest focusing on the critical requirements first :)

Feel free to ask me anything.

Thank you
Bruno

PS: I have mixed feelings regarding Realms integration. I don't know if it integration os fast nor slow but I woudn't risk it if there are tasks with greater priority...

**Comment by teodorciuraru** [comment]
Hello Bruno,

I'm looking forward to contribute. Can you guide me through what you would do as the next step? 

Consider me being an experienced iOS developer. I would instantly look into refactoring, updating the code to Swift 2.1 / 2.2, slowly migrating to Carthage from Cocoapods.

I am looking to contribute as a student in this year's GSoC. We can also chat on Slack or any other messaging platforms.

**Comment by teo029** [comment]
If everything feels fine to you. I can start solving the GitHub issues, but I need to be sure this will count toward a subsequent application.

**Comment by teodorciuraru** [comment]
Oh, I actually like this project! If everything feels fine to you, I can start solving the GitHub issues. :smile:

**Comment by bphenriques** [comment]
Hello @teo029 ,

Glad you like it! Feel free to take a look into the current issues! Currently the big one is integrating with fabric.io and Realms however, you can start with the minor issues (why not a widget or social integration?) :) Any contribution is appreciated! Don't forget to test the app and use unit tests ;)

Submit different PR for different issues and good luck for GSoC 2016! :) 

PS: Or any bugs you can find. Or even updating the pods version.

**Comment by manas15** [comment]
Hello @bphenriques 

I liked your project and would like to contribute on this project as a part of GSOC 2016. I have a small doubt, since this organisation promotes women in code do they only accept women coders or both men and women can contribute on this project ?

Cheers 
Manas Sharma

**Comment by bphenriques** [comment]
Both men and women are accepted :)

**Comment by manas15** [comment]
Thanks @bphenriques

**Comment by manas15** [comment]
@bphenriques 
How can we connect with the mentor for this project. I'm asking this question because i'm not clear about the expectations from the organisation regarding this project.  :)

**Comment by bphenriques** [comment]
@manas15 , look into slide 29 of the google slides link in the first post (social features, gamify, track medicine stock)

**Comment by Dwar3xwar** [comment]
Hi @bphenriques. Just wanted to introduce myself as another student. This is a great example of how technology can help save lives :+1:

**Comment by bphenriques** [comment]
@Dwar3xwar Hello there, welcome! Feel free to ask anything and join us on our slack channel :)

**Comment by Dwar3xwar** [comment]
Whats the slack channel? Can't seem to find it anywhere

**Comment by bphenriques** [comment]
http://systers.io/slack-systers-opensource/

2016-03-07 22:04 GMT+00:00 Dwar3xwar notifications@github.com:

> Whats the slack channel? Can't seem to find it anywhere
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/systers/malaria-app-ios/issues/32#issuecomment-193476426
> .

## 

Os melhores cumprimentos,
Bruno Henriques

**Comment by bphenriques** [comment]
Gentlemen,

I will no longer accept commit messages like:
- "Ready to be reviewed"
- "Final Touches"
- "Pod file issue solved" (what was?)
- "making code readble"
- "removed excessive new lines 2"

I will ask you to either change the commit messages or squash the commits into one saying: "Fix #IssueNumber" or whatever.

For commits I suggest reading more on the matter:
http://chris.beams.io/posts/git-commit/
- Capitalize first letter. 
- Be clear on what it changes it does. 
- Do not end with ".". 
- Present tense. 
  -"Use the imperative mood in the subject line".

Yes, it sucks writing commit messages. I understand and I am sometimes lazy as well. Do it on your own projects. However, when working in teams, it is easier to explain what you did through the commit title (and description) instead of explaining it over through slack (inefficient). Moreover, it is easy to understand what that commit does 5 months later. Yes, it is a growing skill (I am also trying to force myself into doing this) and I strongly believe that it is a important skill to have so I will be strict with this :)

**Write once read many** (WORM)

**Comment by bphenriques** [comment]
https://github.com/systers/malaria-app-ios/blob/develop/InitialRequirements/RequirementsAugustApril2014.docx

**Comment by bphenriques** [comment]
https://www.atlassian.com/git/tutorials/comparing-workflows/feature-branch-workflow (Feature Branch Workflow)
1. Keep your master and develop branch in sync with systers
2. Create a branch from the develop syster's branch: "mistery-feature"
3. Do your changes on your new branch "mistery-feature"
4. Before submitting a pull request check if there are changes from the develop branch
5. Submit a new pull request from "mistery-feature"

**Comment by n0shake** [comment]
Hello @bphenriques. I'm Abhishek, another student looking to contribute :) I really love the work you have done so far.

In your original post, you mention

> have better insight on our users

What kind of insights are you referring to for eg. something like tracking user actions through analytics? Can you expand on this?

**Comment by bphenriques** [comment]
Yeah sure. I think it would be beneficial to track custom events to better
understand what was the sequence of user actions that triggered crash. Not
just stack trace.

For example:
user-id-added-entry-Ok-Abril-19-2015
user-id-modified-entry-to-NotOk-Abril-19-2015
user-id-added-entry-NotOk-Abril-19-2015
<Crash>

Ok: took the pill
NOk: did not take the pill

I think this complements the existing information and gives a better
insight on what it is happening.

2016-03-17 22:54 GMT+00:00 Abhishek Banthia notifications@github.com:

> Hello @bphenriques https://github.com/bphenriques. I'm Abhishek,
> another student looking to contribute :) I really love the work you have
> done so far. In your original post, you mention
> 
> have better insight on our users
> 
> What kind of insights are you referring to for eg. something like tracking
> user actions through analytics? Can you expand on this?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> https://github.com/systers/malaria-app-ios/issues/32#issuecomment-198115087

## 

Os melhores cumprimentos,
Bruno Henriques

**Comment by bphenriques** [comment]
Major change in the first comment.
